### PR TITLE
[Jobs] Show pool health in the wait spinner for pool-targeted launches

### PR DESCRIPTION
**Pull request by lloyd-brown**
## Summary

When `sky jobs launch --pool <pool>` can't dispatch immediately (e.g. no idle replicas), the job sits in PENDING showing only the generic "Waiting for task to start" spinner. The user has no signal whether the pool is fundamentally broken (zero READY workers) or just busy with someone else's work — they have to run `sky jobs pool status` in another terminal to find out.

This PR adds a one-line pool-health summary to that spinner, refreshed each tick, so the situation is visible inline:

```
⠋ Waiting for task to start (status: PENDING). It may take a few minutes.
  Pool 'foo': 1/3 ready (2 provisioning)
  View controller logs: sky jobs logs --controller 42
```

- A small helper `_get_pool_health_summary` in `sky/jobs/utils.py` queries `serve_state.get_replica_infos` and produces strings like `Pool 'foo': 5/11 ready (3 provisioning, 2 starting, 1 failed)`.
- All `FAILED_*` replica statuses (`FAILED`, `FAILED_CLEANUP`, `FAILED_INITIAL_DELAY`, `FAILED_PROBING`, `FAILED_PROVISION`, `UNKNOWN`) collapse to a single `failed` count via `ReplicaStatus.failed_statuses()` so the line stays readable for large/chaotic pools.
- A new `{pool_str}` field is added to `_JOB_WAITING_STATUS_MESSAGE`. When `pool` is unset (non-pool launches), it stays empty, so non-pool behavior is unchanged.
- Any error querying replicas is swallowed; the spinner falls back to the original message.

## Test plan

Renderings across pool scales (verified via `unittest.mock.patch.object(serve_state, 'get_replica_infos', ...)`):

| Scenario | Rendered summary |
|---|---|
| Empty / nonexistent pool | `Pool 'demo-pool': 0 workers` |
| Healthy small pool, all ready | `Pool 'demo-pool': 3/3 ready` |
| Small pool warming up | `Pool 'demo-pool': 1/3 ready (2 provisioning)` |
| Small pool, broken (mixed FAILED_*) | `Pool 'demo-pool': 0/3 ready (3 failed)` |
| Mid pool (mixed) | `Pool 'demo-pool': 5/11 ready (3 provisioning, 2 starting, 1 failed)` |
| Large pool (50) | `Pool 'demo-pool': 30/50 ready (10 provisioning, 5 starting, 3 not_ready, 2 failed)` |
| Chaotic 200-pool | `Pool 'demo-pool': 80/200 ready (50 provisioning, 30 starting, 10 not_ready, 10 pending, 5 shutting_down, 15 failed)` |

End-to-end manual test:

- [ ] `sky jobs launch --pool <healthy-pool> -- echo hi` — confirm spinner shows `Pool '<name>': N/N ready` while waiting, transitions to streaming logs after dispatch.
- [ ] `sky jobs launch --pool <pool-with-no-ready-workers> -- echo hi` — confirm spinner shows `0/N ready (...)` instead of leaving the user in the dark.
- [ ] `sky jobs launch -- echo hi` (no `--pool`) — confirm spinner is byte-identical to pre-change output (no pool line).
- [ ] `format.sh` clean (10.00/10 pylint, mypy, yapf, isort, black all pass).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/skypilot-org/skypilot/pull/9520" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: sky/jobs/utils.py [hunk 1]
```diff
@@ -1195,6 +1196,41 @@ def controller_log_file_for_job(job_id: int,
...
+            if r.status in failed_set:
+                label = 'failed'
+            else:
+                label = r.status.value.lower()
+            other[label] = other.get(label, 0) + 1
+        if other:
+            other_str = ', '.join(f'{c} {label}' for label, c in other.items())
+            return (f'Pool {pool_name!r}: {ready}/{total} ready '
+                    f'({other_str})')
+        return f'Pool {pool_name!r}: {ready}/{total} ready'
```
[the remark this answers is not in the document]
Comment (lloyd-brown) [reply sky/jobs/utils.py]: Took the single-pass and `collections.Counter` suggestions in 5af9f72.

Skipped the alphabetical sort and used **count descending (alphabetical tiebreaker)** instead. Alphabetical would put `failed` near the front in most realistic cases (e.g. `(15 failed, 10 not_ready, 10 pending, 50 provisioning, 5 shutting_down, 30 starting)` for a 200-pool that's mostly provisioning), which misleads the reader on where the bulk of replicas actually are. Count-desc puts the dominant state first — the answer to "what's happening to most replicas right now?" — and is still deterministic.

**Comment by cblmemo** [comment]
According to the scenario you described, should we show how many pool replica is occupied by other jobs as well?

**Comment by lloyd-brown** [comment]
> According to the scenario you described, should we show how many pool replica is occupied by other jobs as well?

Ah yes I like that, will work on adding that!

**Comment by lloyd-brown** [comment]
Good call. Addressed in 4776f3b4 — replaced `N/total ready` with `N/total idle` and added an `occupied` count from `managed_job_state.get_num_alive_jobs(pool=pool_name)`:

```
# Before:
Pool 'foo': 1/3 ready (2 provisioning)

# After (no contention):
Pool 'foo': 1/3 idle (2 provisioning)

# After (contention):
Pool 'foo': 0/3 idle (1 occupied, 2 provisioning)
```

`idle` directly answers "how many workers can my job use right now?" and `occupied` tells the user it's queue contention rather than a broken pool. Occupied is clamped to the ready count to handle race conditions between replica state and job state. Unit tests cover empty pool, all-idle, partial idle/occupied, all-occupied, clamp edge case, and failed-replica collapsing.
